src/wallet/wallet.cpp

```diff
@@ -4002,6 +4002,23 @@ bool CWallet::ApplyMigrationData(MigrationData& data, bilingual_str& error)
             }
         }
     }
+
+    // Persist added address book entries (labels, purpose) for watchonly and solvable wallets
+    auto persist_address_book = [](const CWallet& wallet) {
+        LOCK(wallet.cs_wallet);
+        WalletBatch batch{wallet.GetDatabase()};
```

Comment:
Need to call `batch.TxnBegin()` and `batch.TxnCommit()` to make this a real batch (yes, the `WalletBatch` name is misleading).

Extra note:
It's basically what have done in #25297. Batch entire processes where is possible.
Which.. should revive it soon and see how the current wallet behaves.